pkg/agent/utils.go

```diff
@@ -586,3 +586,25 @@ func addFeatureGateString(featureGates string, key string, value bool) string {
 	}
 	return strings.Join(pairs, ",")
 }
+
+// ParseCSEMessage parses the raw CSE output
+func ParseCSEMessage(message string) (*datamodel.InstanceViewCSEStatus, error) {
+	var cseStatus datamodel.InstanceViewCSEStatus
+	start := strings.Index(message, "[stdout]") + len("[stdout]")
+	end := strings.Index(message, "[stderr]")
+	if end > start {
+		rawInstanceViewInfo := message[start:end]
+		err := json.Unmarshal([]byte(rawInstanceViewInfo), &cseStatus)
+		if err != nil {
+			// Regex "InstanceErrorCode=" is used to parse the error.
+			return nil, fmt.Errorf("CSE has invalid message=%s, %s=%s", message, InstanceErrorCode, CSEMessageUnmarshalError)
```

Comment:
why do not we return CSEMessageUnmarshalError as the third  output, this way we don't need that regex.

func ParseCSEMessage(message string) (*datamodel.InstanceViewCSEStatus, string, error) {

Also create a new error struct that contains the error code and error message?